packages/coverage-v8/src/provider.ts

```diff
@@ -260,7 +261,13 @@ export class V8CoverageProvider extends BaseCoverageProvider implements Coverage
       }
 
       const coverages = await Promise.all(chunk.map(async (filename) => {
-        const { source } = await this.getSources(filename.href, transformResults)
+        const transformResult = await this.ctx.vitenode.transformRequest(filename.pathname)
+
+        // Ignore empty files, e.g. files that contain only typescript types and no runtime code
+        if (transformResult && await isFileEmpty(transformResult.code))
```

Comment:
Why do we really need to parse `transformResult` AST? It's always JavaScript, so it will never have type declarations 🤔

Reply:
I guess we could make this decision based on `mappings` as well. Empty mappings would indicate that the file contains no runtime code.

```json
{
  "code": "\n\n//# sourceMappingSource=vite-node\n//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInR5cGVzLnRzIl0sInNvdXJjZXNDb250ZW50IjpbImV4cG9ydCB0eXBlIEZpcnN0ID0gXCJpZ25vcmVcIiB8IFwidGhpc1wiIHwgXCJmaWxlXCI7XG5cbmV4cG9ydCB0eXBlIFNlY29uZCA9IFwiaWdub3JlXCIgfCBcInRoaXNcIiB8IFwiZmlsZVwiO1xuIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiIifQ==\n",
  "map": {
    "version": 3,
    "sources": [
      "types.ts"
    ],
    "sourcesContent": [
      "export type First = \"ignore\" | \"this\" | \"file\";\n\nexport type Second = \"ignore\" | \"this\" | \"file\";\n"
    ],
    "names": [],
    "mappings": ""
  },
  "deps": [],
  "dynamicDeps": []
}
```

Reply:
Can we trim the code an check that it doesn't start with `//# sourceMappingSource`?

Reply:
It would work for most cases but there can be edge cases that won't fall into that check, e.g.:

```ts
// Some comment here @preserve
export type First = "ignore" | "this" | "file";
```

This will transform to 
```js
// Some comment here @preserve

//# sourceMappingSource=vite-node
...
```

At the moment I can't think of other such cases but I guess there could be. AST based check would always be accurate. I'll change this to `transformResult?.code.trim().startsWith('//# sourceMappingSource=')` - it should be accurate enough.

Reply:
Maybe we can use something like https://www.npmjs.com/package/strip-literal then?

Reply:
That looks good. We could use `stripLiteral(transformResult.code).trim() === ''` check here.